src/op_mode/nat.py

```diff
@@ -126,21 +126,33 @@ def _get_formatted_output_rules(data, direction, family):
                     elif my_dict['field'] == 'daddr':
                         daddr = f'{op}{my_dict["prefix"]["addr"]}/{my_dict["prefix"]["len"]}'
                     elif my_dict['field'] == 'sport':
-                        # Port range or single port
-                        if jmespath.search('set[*].range', my_dict):
-                            sport = my_dict['set'][0]['range']
-                            sport = '-'.join(map(str, sport))
-                        else:
-                            sport = my_dict.get('set')
-                            sport = ','.join(map(str, sport))
+                        # Get all configured sport ranges or single ports
+                        for index, port in enumerate(my_dict['set']):
```

Comment:
This code path is more or less duplicated below only for dport. It would make sense to have this as an inlined function inside `_get_formatted_output_rules()`: like:

```python3
def _get_formatted_output_rules(data, direction, family):
    def _range_to_port_string(my_dict):
        for index, port in enumerate(my_dict['set']):
        ...

```

Reply:
Makes sense, will do that and report back.